src/LanguageServer/Impl/Sources/ReferenceSource.cs

```diff
@@ -146,10 +146,20 @@ internal sealed class ReferenceSource {
 
             return analysisTasks.Count > 0;
         }
-        
+
         private ILocatedMember GetRootDefinition(ILocatedMember lm) {
-            for (; lm.Parent != null; lm = lm.Parent) { }
-            return lm;
+            if (!(lm is IImportedMember im) || im.Parent == null) {
+                return lm;
+            }
+
+            var parent = im.Parent;
+            for (; parent != null;) {
+                if (!(parent is IImportedMember im1) || im1.Parent == null) {
```

Comment:
The condition here is slightly different, I assume there's no way intended for a variable to not be the root but not implement IImportedMember? (Since before a `null` parent was the indicator, but now you could technically end early.)

Reply:
Basically it looks for a root that does not have parent. In new code only imported member has parent, so yes, member that is not imported cannot have the parent.